sbndaq-artdaq/Generators/Common/CAENV1730Readout_generator.cc

```diff
@@ -1729,6 +1743,22 @@ bool sbndaq::CAENV1730Readout::readSingleWindowFragments(artdaq::FragmentPtrs &
 	
 	ts_frag = (t_truetriggertime*8); //in 1ns ticks
       }
+      else if(fUseTimeTagShiftForTimeStamp){
+	const auto TTT = uint32_t {header->triggerTimeTag};
+	
+	using namespace boost::gregorian;
+	using namespace boost::posix_time;
+	
+	ptime t_now(second_clock::universal_time());
+	ptime time_t_epoch(date(1970,1,1));
+	time_duration diff = t_now - time_t_epoch;
+	uint32_t t_offset_s = diff.total_seconds();
+	uint64_t t_offset_ticks = diff.total_seconds()*125000000; //in 8ns ticks
+	uint64_t t_truetriggertime = t_offset_ticks + TTT;
+	TLOG_ARB(TMAKEFRAG,TRACE_NAME) << "time offset = " << t_offset_ticks << " ns since the epoch"<< TLOG_ENDL;
+	
+	ts_frag = (t_truetriggertime*8) - fCAEN.recordLength * fCAEN.postPercent / 100 * 2; //in 1ns ticks
+      }
```

Comment:
Expanding on my comment above, and looking at the fragment generation in `readSingleWindowFragments`:

- The time of the fragment (`ts_frag`) is built in this way: last **full seconds** from `t_now` (since epoch) +  nanoseconds of the TTT - length of post-trigger buffer in ns. 
- _Note_: both the TTT and the buffer length are < 1s, so they are not changing the "seconds" part of the timestamp.
- The original `TTT` remains available in the event header, while `ts_frag` is saved in the fragment metadata. So both values can be accessed in the offline within the same fragment.
- If you remove the **full seconds** from `ts_frag` (which would correspond to `t_offset_ticks` in the above code snippet) and you also subtract the TTT (read from the header), you can recover the post-trigger length of the buffer.

Given this, it seems that saving the `fCAEN.postPercent` in the metadata is superflous in this case as well.